### DISCUSSION TITLE
Use the _like filter rule to improve searches and autocomplete

### DISCUSSION THREAD

**Discussion by TheoCouss** [directus/directus#20269] [Feature Requests]
### Summary

We sometimes need to filter data based on user input for small content listings, blog searches, etc. Users often make typing mistakes, and we don't want to rely on any third-party software to address this. We typically use _icontains to find a substring in the database, but this method is too restrictive.

### Basic Example

Adding a filter rule named _like and _nlike that behaves like the %LIKE% clause in MySQL
```json
{
  "name": {
    "_like": "lrd"
  }
}
```
This would match all records where the name like the string "lrd", e.g. "lord"

### Motivation

this aims to streamline blog and product searches, content listings, autocompletes, etc., all natively without relying on third-party software.

### Detailed Design

```json
{
  "name": {
    "_like": "lrd"
  }
}
```
```json
{
  "name": {
    "_nlike": "lrd"
  }
}
```

### Must Have:
- _like and _nlike filter rule this quite simple

### Drawbacks

This can be achieved with a custom endpoint, but it won't support all databases. This will add some complexity to the filter layer, and not everyone needs it

### Alternatives

Using a Elasticsearch ou another search engines, making an extension

### Adoption Strategy

This won't be a breaking change because nothing gets removed from the api. I think this can be a faster way to make simple apps without much requirements.

### Unresolved Questions

Is this function compatible with all databases ?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
